client/galaxy/scripts/mvc/ui/ui-select-default.js

```diff
@@ -186,6 +187,13 @@ var View = Backbone.View.extend({
 
     /** Update all available options at once
     */
+    add: function( options, sorter ) {
+        _.each( this.model.get( 'options' ), function( v ) {
+            !_.findWhere( options, v ) && options.push( v );
+        });
```

Comment:
If `options` and `model.get( 'options' )` are both arrays, can't you use `_.union` here? http://underscorejs.org/#union